src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -79,65 +87,82 @@ public virtual void FileWorkItems(string sarifLogFileContents)
+                if (optionallyEmittedData != OptionallyEmittedData.None)
                 {
-                    partitionFunction = (result) => result.ShouldBeFiled() ? "Include" : null;
-                    break;
+                    var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
+                    dataInsertingVisitor.Visit(sarifLog);
                 }
-                case SplittingStrategy.PerResult:
+
+                SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
+
+                if (splittingStrategy == SplittingStrategy.None)
                 {
-                    partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
-                    break;
+                    FileWorkItemsHelper(sarifLog, this.FilingContext, this.FilingClient);
+                    return;
                 }
-                default:
+
+                IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                PartitionFunction<string> partitionFunction = null;
+
+                Stopwatch splittingStopwatch = Stopwatch.StartNew();
+                
+                switch (splittingStrategy)
                 {
-                    throw new ArgumentOutOfRangeException($"SplittingStrategy: {splittingStrategy}");
+                    case SplittingStrategy.PerRun:
+                    {
+                        partitionFunction = (result) => result.ShouldBeFiled() ? "Include" : null;
+                        break;
+                    }
+                    case SplittingStrategy.PerResult:
+                    {
+                        partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
+                        break;
+                    }
+                    default:
+                    {
+                        throw new ArgumentOutOfRangeException($"SplittingStrategy: {splittingStrategy}");
+                    }
                 }
-            }
 
-            var partitioningVisitor = new PartitioningVisitor<string>(partitionFunction, deepClone: false);
-            partitioningVisitor.VisitSarifLog(sarifLog);
+                var partitioningVisitor = new PartitioningVisitor<string>(partitionFunction, deepClone: false);
+                partitioningVisitor.VisitSarifLog(sarifLog);
 
-            logsToProcess = new List<SarifLog>(partitioningVisitor.GetPartitionLogs().Values);
+                logsToProcess = new List<SarifLog>(partitioningVisitor.GetPartitionLogs().Values);
 
-            for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
-            {
-                SarifLog splitLog = logsToProcess[splitFileIndex];
-                FileWorkItemsHelper(splitLog, this.FilingContext, this.FilingClient);
+                var logsToProcessMetrics = new Dictionary<string, object>
+                {
```

Comment:
Thoughts on if I should look at building a class that creates the dictionary rather than in-lining?

Reply:
I suspect we'll want a method that accepts an EventId and a list of environment variables and then creates the dictionary and sends the telemetry. Mostly because I suspect we'll want to capture environment variables from multiple environments. i.e. not only ADO build agents.

---
In reply to: [395897876](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r395897876) [](ancestors = 395897876)

Reply:
Ok, but will I want that method in a new class or should I put it in one you already created. I was trying to decide if it fit in anything you already added.

Reply:
I suspect a new class. I think we'll need a TelemetryHelpers class (or similarly named) to hold all the common telemetry functions. 

---
In reply to: [395908403](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r395908403) [](ancestors = 395908403)